blockchain/serviceMetadata.go

```diff
@@ -20,44 +20,50 @@ type ServiceMetadata struct {
 	DisplayName                string   `json:"display_name"`
 	Encoding                   string   `json:"encoding"`
 	ServiceType                string   `json:"service_type"`
-	PaymentExpirationThreshold *big.Int `json:"payment_expiration_threshold"`
+    Groups                     []OrganizationGroup `json:"groups"`
 	ModelIpfsHash              string   `json:"model_ipfs_hash"`
 	MpeAddress                 string   `json:"mpe_address"`
-	Pricing                    struct {
-		PriceModel  string `json:"price_model"`
-		PackageName string `json:"package_name"`
-		//Price in cogs has been retained only to support backward compatibility
-		PriceInCogs *big.Int `json:"price_in_cogs"`
-		Details     []struct {
-			ServiceName   string `json:"service_name"`
-			MethodPricing []struct {
-				MethodName  string   `json:"method_name"`
-				PriceInCogs *big.Int `json:"price_in_cogs"`
-			} `json:"method_pricing"`
-		} `json:"details"`
-	} `json:"pricing"`
-	Groups []struct {
-		GroupName      string `json:"group_name"`
-		GroupID        string `json:"group_id"`
-		PaymentAddress string `json:"payment_address"`
-	} `json:"groups"`
-	Endpoints []struct {
-		GroupName string `json:"group_name"`
-		Endpoint  string `json:"endpoint"`
-	} `json:"endpoints"`
-	daemonReplicaGroupIDString string
-	daemonReplicaGroupID       [32]byte
-	daemonGroupName            string
-	daemonEndPoint             string
-	recipientPaymentAddress    common.Address
+
 	multiPartyEscrowAddress    common.Address
+	defaultPricing Pricing
+
+	defaultGroup                OrganizationGroup
+}
+
+type OrganizationGroup struct {
+	Endpoints []string `json:"endpoints"`
+	GroupID   string   `json:"group_id"`
+	GroupName      string  `json:"group_name"`
+	Pricing   []Pricing  `json:"pricing"`
+}
+type Pricing struct {
+	PriceModel  string `json:"price_model"`
+	PriceInCogs *big.Int    `json:"price_in_cogs,omitempty"`
+	PackageName string `json:"package_name,omitempty"`
+	Default     bool   `json:"default,omitempty"`
+	PricingDetails []PricingDetails `json:"details,omitempty"`
 }
 
+type PricingDetails struct {
+	ServiceName   string               `json:"service_name"`
+	MethodPricing []MethodPricing `json:"method_pricing"`
+}
+type MethodPricing struct {
+	MethodName  string `json:"method_name"`
+	PriceInCogs *big.Int    `json:"price_in_cogs"`
+}
+
+
+
 func getRegistryAddressKey() common.Address {
 	address := config.GetRegistryAddress()
 	return common.HexToAddress(address)
 }
 
```

Comment:
Make sure that where ever we have address is in checksum format. Some comparisons might fail.